shelley-ma/formal-spec/value.tex

```diff
@@ -1,77 +1,239 @@
+fees, rewards, treasury, and the proof of stake protocol. For this reason, not
+all occurrences of $\Coin$ inside a transaction or in the ledger state can or
+should be replaced with the abstract $\ValMonoid$.
 
-\begin{figure*}[t!]
-  \emph{Abstract Types}
-  %
+%%
+%% Figure Coin Functions
+%%
+\begin{figure}[htb]
   \begin{align*}
-    \var{aname} \in& ~\AssetName & \text{Asset Names}
+      & \fun{zero} ~=~ 0
+      \nextdef
+      %
+      & \fun{coin} ~=~ \fun{id}_{\Coin}
+      \nextdef
+      %
+      & \fun{inject} ~=~ \fun{id}_{\Coin}
+      \nextdef
+      %
+      & \fun{modifyCoin} ~=~ \fun{id}_{(\Coin \to \Coin)}
+      \nextdef
+      %
+      & \fun{size}~\wcard ~=~ 0
+      \nextdef
+      %
+      & \fun{pointwise} ~=~ \fun{id}_{(\Integer \to \Integer \to \Bool)}
   \end{align*}
-  %
+  \caption{$\ValMonoid$ structure with $\Coin$}
+  \label{fig:coin}
+\end{figure}
+
+
+
+\noindent \textbf{Relation between $\fun{inject}$ and $\fun{coin}$.}
+When composed as
+
+\[\fun{coin} \circ \fun{inject} = \fun{id}_{\Coin}\]
+
+these functions give the identity on $\Coin$, but composed in the opposite order as
+
+\[\fun{inject} \circ \fun{coin}\]
+
+they return a $\ValMonoid$ containing only the Ada tokens contained in the original $\ValMonoid$.
+
+\textbf{Fungible and non-fungible tokens.}
+Recall that a token is said to be fungible at a specific time (slot $s$, for example) if
+there are two or more tokens in existence on the ledger that are indistinguishable from
+each other, ie. they have the exact same identifier. Tokens with the same
+identifier are fungible with each other, but not with tokens with a different identifier.
+
+If there exists only a single
+token with a given identifier, it is said to be unique (or non-fungible).
+The usual meaning of \emph{non-fungible token} implies, additionally to the
+requirement that a token is unique on the ledger at a given time, that
+it is programmatically guaranteed to remain unique indefinitely. This
+type of guarantee can be encoded in a minting policy.
+
+
+\subsection{Multi-assets and $\Value$ in the Mary Era}
+
+Terms of the $\Value$ type represent heterogeneous collections of assets,
+both user-defined, and Ada. In place of the abstract $\ValMonoid$ structure,
+this type is used in the concrete Mary era ledger.
+This section describes the type $\Value$, the operations required on
+it, its relation to $\Coin$. It also as gives the definitions of the
+$\ValMonoid$ functions on $\Value$.
+
+Figure \ref{fig:defs:value} introduces the basic types that make up the $\Value$ type,
+as well as defining the functions required to for $\Value$ to be a $\ValMonoid$.
+
+\begin{figure*}[t!]
   \emph{Derived types}
   %
   \begin{equation*}
     \begin{array}{r@{~\in~}l@{\qquad=\qquad}lr}
+      \var{aname} & ~\AssetName & \mathsf{ByteString} \\
       \var{pid} & \PolicyID & \ScriptHash \\
+      \var{adaID} & ~\AdaIDType & \{~\mathsf{AdaID}~\} \\
```

Comment:
This type is unused

Reply:
it is unused because it does not exist the ledger implementation. It is used in the explanation of how to implement Ada with this example of `ValMonoid`

Reply:
The only places where the string `AdaIDType` exist in the source files are here, and at the definition of the `AdaIDType` macro. So this type is currently not mentioned anywhere except here

Reply:
i added a brief description of it

Reply:
I understand now. It's still unused though, and to fix that, you have to replace `AdaID` in the line below by `AdaIDType`